### PR TITLE
MOB-40 - onesignal blog post

### PR DESCRIPTION
**Pull request by mateusz-bajorek**

### CODE REVIEW

File: _posts/2017_01_20_onesignal.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+- Travis CI
+authors:
+- Mateusz Bajorek
+intro: "Developer Contributions Will Help Make OneSignal-Java-SDK Even Better"
+image: onesignal.png
+---
+{% include JB/setup %}
+
+## Introduction
+We at the CurrencyFair have recently released an update to our system that enables customers to receive push notifications
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: remove `the`

File: _posts/2017_01_20_onesignal.md [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+authors:
+- Mateusz Bajorek
+intro: "Developer Contributions Will Help Make OneSignal-Java-SDK Even Better"
+image: onesignal.png
+---
+{% include JB/setup %}
+
+## Introduction
+We at the CurrencyFair have recently released an update to our system that enables customers to receive push notifications 
+about deposit being cleared or Auto Transaction being completed - you can read more about those features on
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: `a Deposit` .. `an Auto Transaction`

File: _posts/2017_01_20_onesignal.md [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+
+## Introduction
+We at the CurrencyFair have recently released an update to our system that enables customers to receive push notifications 
+about deposit being cleared or Auto Transaction being completed - you can read more about those features on 
+[our marketing blog](https://www.currencyfair.com/blog/sms-and-push-notifications/). For those push notifications we 
+decided to use [OneSignal](https://onesignal.com) in the backend - a high volume, cross platform push notification 
+delivery provider. Their API is really nice and they offer SDKs for many, many development environments. Go on, check 
+them out too! Adobe, Uber and Line, among many others, also use them!
+
+One of the drawbacks when using OneSignal was however lack of native Java library to programmatically compose and send
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: `a lack of a`

File: _posts/2017_01_20_onesignal.md [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+
+
+### Continuous Deployment
+After reading many tutorials and after many tries we finally nailed our own version of deploying our releases: we use
+[Travis CI](https://travis-ci.org) for building and [Central Repository](http://central.sonatype.org/) to host the 
+artifacts. It wasn't an easy task so here are the steps you would want to follow to reach same objectives.
+
+1. [Create a Jira Account at Sonatype](https://issues.sonatype.org/secure/Signup!default.jspa).
+
+2. [Create a project ticket](https://issues.sonatype.org/secure/CreateIssue.jspa?issuetype=21&pid=10134).
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: Should this link remain? Its not accessible unless you have an account

File: _posts/2017_01_20_onesignal.md (hunk 3 above)
Comment (JasonGermaine) [inline _posts/2017_01_20_onesignal.md]: I would word this as
```
However, one of the drawbacks when using OneSignal was the lack of a native...
```

File: _posts/2017_01_20_onesignal.md [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+push notifications to customers. And so we decided to write our own and, later, publish it as Open Source! You can find
+the source code [here](https://github.com/CurrencyFair/OneSignal-Java-SDK).
+
+## Technicalities
+### Development
+OneSignal provides a well-documented [REST API](https://documentation.onesignal.com/reference). We decided to use 
+[OpenFeign](https://github.com/OpenFeign/feign) together with [FasterXML's jackson](https://github.com/FasterXML/jackson) to 
+communicate with their endpoints.
+
+#### Why feign?
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: `Feign` (capital F) seems to be how they refer to it on their github page

File: _posts/2017_01_20_onesignal.md [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+## Introduction
+We at the CurrencyFair have recently released an update to our system that enables customers to receive push notifications 
+about deposit being cleared or Auto Transaction being completed - you can read more about those features on 
+[our marketing blog](https://www.currencyfair.com/blog/sms-and-push-notifications/). For those push notifications we 
+decided to use [OneSignal](https://onesignal.com) in the backend - a high volume, cross platform push notification 
+delivery provider. Their API is really nice and they offer SDKs for many, many development environments. Go on, check 
+them out too! Adobe, Uber and Line, among many others, also use them!
+
+One of the drawbacks when using OneSignal was however lack of native Java library to programmatically compose and send
+push notifications to customers. And so we decided to write our own and, later, publish it as Open Source! You can find
```
Comment (JasonGermaine) [inline _posts/2017_01_20_onesignal.md]: I think the `,` don't read very well here. Try:
```
And so we decided to write our own and ultimately publish it as Open Source
```

File: _posts/2017_01_20_onesignal.md (hunk 4 above)
Comment (mateusz-bajorek) [inline _posts/2017_01_20_onesignal.md]: Yes, after they create an account they would always be able to come back to the article and click the link again.

File: _posts/2017_01_20_onesignal.md [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+profile. Go to Accounts page and select the repo that you would like to be continuously integrated.
+
+10. Open settings of your repo in Travis (More options -> Settings), you can configure there a couple of things, we are 
+interested in Environment Variables.
+
+11. Add `SONATYPE_USERNAME` with the username you registered your Sonatype Jira account with and `SONATYPE_PASSWORD` with
+the Sonatype Jira password.
+
+12. Add `.travis.yml` to the root of your project. It is very important that you override `install` instruction from 
+`mvn` to use the `mvnsettings.xml` you have created a bit ago.
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: `a short time ago` might read better

File: _posts/2017_01_20_onesignal.md [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+            git config user.email "build@travis-ci.org"
+            git remote add upstream "https://$GITHUB_API_TOKEN@github.com/CurrencyFair/OneSignal-Java-SDK.git"
+            git fetch upstream
+            git add pom.xml
+            git commit -m "[Travis-CI] next snapshot version"
+            git push --quiet upstream master > /dev/null 2>&1
+        fi
+
+    Let's go through this script step by step:
+    * as we don't use maven deployment plugin we introduce `increment_version_to_next_snapshot` function to iterate the
```
Comment (garethpelly) [inline _posts/2017_01_20_onesignal.md]: `introduced the`

**Comment by garethpelly** [comment]
Well written blog post!

**Pull request merged**